# BacSpecies
A python script to predict the species of bacteria isolate using the 16S rRNA seuquence
# Introduction
To know the species of bacteria isolate is the first step after successful isolation, it's the basement to study the characteristics of a isolate, previously, there are a web-based species predict tool, SpeciesFinder, could do this job, here, I write this tools to do the same job based on two reasons. 1. This kind of alignment is very basic, just use a single blastn, and a no-very-difficult logic to handle the blast result, it's very suitable for a beginner like me to get familiar with python; 2. Multiple methods to handle same problems is very useful, could expand our choice to deal with different situations.
This is a pure python script, could direct work in a terminal, or be a part of other bioinformation software, for example, we can combine this script with some MLST, serotyping software, add some if...else... logic, after the species was predict, we can directly guide the program to do some prediction specific for that species.
The reference_database is generated by collected all 16S rRNA nucleotide sequence from NCBI RefSeq database, total 22000+ sequence were collected.
# External Dependencies
BLAST+
# Usage
```
BacSpecies [-i] [-o] [-r] [-t] [-v]
Input and Output:
  -i, --input             Input FASTA file
  -o, --output            Output directory
  -r, --refs              fasta file contain all 16S rRNA sequence from NCBI
  -t, --threads           Threads to use for BLAST searches
  -v, --version           Show version number and exit
```
# Quick usage
``` Python
python BacSpecies.py -i *.fasta 
```
